qa/tasks/cephfs/test_volumes.py

```diff
@@ -2819,6 +2819,64 @@ def test_authorize_deauthorize_subvolume(self):
         self._fs_cmd("subvolume", "rm", self.volname, subvolume, "--group_name", group)
         self._fs_cmd("subvolumegroup", "rm", self.volname, group)
 
+    def test_subvolume_deauthorize_with_shared_key(self):
+        """
+        That mon caps are preserved when one cephx key authorized on multiple
+        subvolumes is deauthorized on any of those.
+        """
+        subvolume1 = self._gen_subvol_name()
+        subvolume2 = self._gen_subvol_name()
+        group = self._gen_subvol_grp_name()
+        authid = "alice"
+
+        # create group
+        self._fs_cmd("subvolumegroup", "create", self.volname, group)
+
+        # create subvolumes
+        self._fs_cmd("subvolume", "create", self.volname, subvolume1, "--group_name", group)
+        self._fs_cmd("subvolume", "create", self.volname, subvolume2, "--group_name", group)
+
+        # authorize alice authID read-write access to both subvolumes
+        self._fs_cmd("subvolume", "authorize", self.volname, subvolume1, authid,
+                     "--group_name", group)
+        self._fs_cmd("subvolume", "authorize", self.volname, subvolume2, authid,
+                     "--group_name", group)
+
+        # verify autorized-id has access to both subvolumes
+        expected_auth_list = [{'alice': 'rw'}]
+        auth_list1 = json.loads(self._fs_cmd('subvolume', 'authorized_list', self.volname, subvolume1, "--group_name", group))
+        self.assertEqual(expected_auth_list, auth_list1)
+        auth_list2 = json.loads(self._fs_cmd('subvolume', 'authorized_list', self.volname, subvolume2, "--group_name", group))
+        self.assertEqual(expected_auth_list, auth_list2)
+
+        # check mon caps for authid
+        expected_mon_caps = 'allow r'
+        full_caps = self._raw_cmd("auth", "get", "client.alice", "--format=json-pretty")
```

Comment:
@ebocchi - this call needs to be wrapped around `json.loads()`, else accessing the json would fail as it fails in

https://pulpito.ceph.com/vshankar-2024-10-30_07:41:03-fs-wip-vshankar-testing-20241029.045257-debug-testing-default-smithi/7973229/

```
024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:======================================================================
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:ERROR: test_subvolume_deauthorize_with_shared_key (tasks.cephfs.test_volumes.TestSubvolumes)
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:That mon caps are preserved when one cephx key authorized on multiple
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_73c971683afc1322bab0582d2b64477919ac88b7/qa/tasks/cephfs/test_volumes.py", line 2855, in test_subvolume_deauthorize_with_shared_key
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:    self.assertEqual(expected_mon_caps, full_caps[0]['caps']['mon'])
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:TypeError: string indices must be integers
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:
2024-10-31T01:31:12.006 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
```

Reply:
Sorry, missed that.
It should be fixed with 8c28c1e1f15feded8f91c030acf2d86543561c7a.

Reply:
I'll put this to test. The commit can be squashed later when the change is ready to merge. Thx @ebocchi